ncat/ncat_listen.c

```diff
@@ -360,6 +359,7 @@ static int ncat_listen_stream(int proto)
                 logdebug("fd %d is ready\n", i);
 
 #ifdef HAVE_OPENSSL
+            struct fdinfo *fdi = NULL;
```

Comment:
C variable declarations have to be at the start of a block. GCC allows this as a GNU extension, but it could cause problems on other compilers. Just move it 3 lines down to the first line within the `if` block.